packages/fast-colors/src/colorlib/colorRGBA64.ts

```diff
@@ -0,0 +1,131 @@
+// tslint:disable:member-ordering
+import {
+    clamp,
+    denormalize,
+    getHexStringForByte,
+    roundToPrecisionSmall,
+} from "./mathUtils";
+
+export class ColorRGBA64 {
+    public static fromObject(data: any): ColorRGBA64 | null {
+        if (data && !isNaN(data.r) && !isNaN(data.g) && !isNaN(data.b)) {
+            if (!isNaN(data.a)) {
+                return new ColorRGBA64(data.r, data.g, data.b, data.a);
+            } else {
+                return new ColorRGBA64(data.r, data.g, data.b, 1);
+            }
+        }
+        return null;
+    }
+
+    constructor(red: number, green: number, blue: number, alpha: number) {
+        this.r = red;
+        this.g = green;
+        this.b = blue;
+        this.a = alpha;
+    }
+
+    // Scaled to the range [0.0 , 1.0]. Values outside this range are allowed but any methods that convert or tostring the values will also be clamped
+    public readonly r: number;
+    public readonly g: number;
+    public readonly b: number;
+    public readonly a: number;
+
+    public equalValue(rhs: ColorRGBA64): boolean {
+        return (
+            this.r === rhs.r && this.g === rhs.g && this.b === rhs.b && this.a === rhs.a
+        );
+    }
+
+    // #RRGGBB
+    public toStringHexRGB(): string {
+        return (
+            "#" +
+            getHexStringForByte(denormalize(this.r, 0.0, 255.0)) +
+            getHexStringForByte(denormalize(this.g, 0.0, 255.0)) +
+            getHexStringForByte(denormalize(this.b, 0.0, 255.0))
+        );
+    }
+
+    // #RRGGBBAA
+    public toStringHexRGBA(): string {
+        return (

```

Comment:
you can simplify this to `return this.toStringHexRGB() + getHexStringForByte(denormalize(this.a, 0.0, 255.0))`